docs/proposals/secrets-proposal.md

```diff
@@ -0,0 +1,133 @@
+## Proposal for supporting secrets in OpenCompose
+
+### Kubernetes Secrets
+
+A very simple secret object in Kubernetes looks like this -
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: kubesec
+type: Opaque
+data:
+  username: <base64 encoded value>
+  password: <base64 encoded value>
+```
+
+Note that, as seen above, one secret can have multiple `data` values.
+
+---
+
+#### Creating a new secret:
+
+To create a new secret, a top level `secrets` key can be used as follows:
+
+ ```yaml
+version: "0.1-dev"
+
+services:
+   name: foobar
+   containers:
+   - image: foo/bar:tag
+     ports:
+     - port: 8080:80
+     env:
+     - foo=bar
+     - travis_creds:
+         secret: ci_secret/travis
+     mounts:
+     - mountPath: /etc/jenkins.conf
+       secret: ci_secret/jenkins
+
+secrets:
+    ci_secret:
```

Comment:
as discussed in other comments here, keys should have a predefined name and shouldn't store data that users inputs.

I would do it like this so  its consistent with other top level sections like `services` and `volumes`:
```
secrets:
  - name: ci_secret
    data:
       - name: jenkins
         value: stringpasswrd
       - name: travis
         file: /etc/travis.passwd
       - name: password
         valueBase64: YXNmZA0KYXNkZg==
```

This is consistent with other parts of OpenCompose spec.
`value`, `file` and `valueBase64` can't be used together in one data field.
`valueBase64` is there so the user can create secrets that contain non-string data (for example binary).